frontend/src/features/housingCompany/HousingCompanyBuildingsPage.tsx

```diff
@@ -0,0 +1,183 @@
+import React, {useState} from "react";
+
+import {Fieldset, IconCrossCircle} from "hds-react";
+import {useParams} from "react-router-dom";
+import {useImmer} from "use-immer";
+
+import {
+    useCreateBuildingMutation,
+    useGetHousingCompanyDetailQuery,
+    useRemoveBuildingMutation,
+} from "../../app/services";
+import {
+    ConfirmDialogModal,
+    FormInputField,
+    NavigateBackButton,
+    SaveButton,
+    SaveDialogModal,
+} from "../../common/components";
+import {IBuildingWritable} from "../../common/models";
+
+const blankForm: IBuildingWritable = {
+    real_estate_id: null,
+    address: {
+        street_address: "",
+    },
+    building_identifier: "",
+};
+
+const HousingCompanyBuildingsPage = (): JSX.Element => {
+    const params = useParams() as {readonly housingCompanyId: string};
+    const [isConfirmModalVisible, setIsConfirmModalVisible] = useState(false);
+    const [isEndModalVisible, setIsEndModalVisible] = useState(false);
+    const [buildingIdToRemove, setBuildingIdToRemove] = useState<string | null>();
+    const [realEstateIdToRemove, setRealEstateIdToRemove] = useState<string | null>();
```

Comment:
Since these are _always_ set together, Instead of two separate states, I'd use a single state for this:

e.g.: 
```
setBuildingToRemove(
  {
    building: building.id,
    realEstate: realEstate.id
  }
);
```